### PR TITLE
Fail a11y and internal tests correctly

### LINKED ISSUE #744: Accessibility failures causes a weird output and an error after all the (usually) final cucumber message
When we gather a11y failures for a Scenario, we wait till the `After()` to fail the test. The way we fail the test is by setting the `scenario.result.status` to `FAILED`. When we do that, cucumber expects to find a failure message (`.message`) in its own internal steps objects (discovered by exploring their code in test_case_attempt_formatter), so instead of failing in a way we recognize, we get something like the following:

```
1) Scenario: My test # test.feature:line
   ✖ Before # steps.js:line
       undefined
   ✔ Given I start the interview at "all_tests" # steps.js:line
   ✔ And I check all pages for accessibility issues # steps.js:line
   ✔ And another step # steps.js:line
   ✔ And another step # steps.js:line
   ✔ After # steps.js:line

1 scenario (1 failed)
4 steps (4 passed)
0m17.605s (executing steps: 0m17.496s)
/Users/repo/node_modules/indent-string/index.js:11
    throw new TypeError(
          ^

TypeError: Expected `input` to be a `string`, got `undefined`
    at module.exports (/Users/repo/node_modules/indent-string/index.js:11:9)
    at formatStep (/Users/repo/node_modules/@cucumber/cucumber/lib/formatter/helpers/test_case_attempt_formatter.js:87:48)
    at /Users/repo/node_modules/@cucumber/cucumber/lib/formatter/helpers/test_case_attempt_formatter.js:101:17
    at Array.forEach (<anonymous>)
    at formatTestCaseAttempt (/Users/repo/node_modules/@cucumber/cucumber/lib/formatter/helpers/test_case_attempt_formatter.js:100:22)
    at formatIssue (/Users/repo/node_modules/@cucumber/cucumber/lib/formatter/helpers/issue_helpers.js:25:94)
    at /Users/repo/node_modules/@cucumber/cucumber/lib/formatter/summary_formatter.js:55:48
    at Array.forEach (<anonymous>)
    at SummaryFormatter.logIssues (/Users/repo/node_modules/@cucumber/cucumber/lib/formatter/summary_formatter.js:54:16)
    at SummaryFormatter.logSummary (/Users/repo/node_modules/@cucumber/cucumber/lib/formatter/summary_formatter.js:41:18)

Node.js v18.17.0
```

Adding a `scenario.status.message` does seem to remove that error that appears after the tests complete, though the `x` still appears in the same place. That says, the `undefined` gets changed to our message. I'm not sure we should keep manipulating the internals of the tests this way, though, and while our output looks ok the cucumber output is a bit confusing.

[Also, I think tests failing because of accessibility issues may not actually fail the whole test suite, so we should check that.]

A different way to handle this would be to create a new Step to check for the passing of all the accessibility checks, e.g. `And all the pages were accessible`. This would be more confusing to the user, unfortunately, but I can't think of another way to handle it that fits into the constraints cucumber presents.

Tangentially: This will also be a problem for some of our internal test failures because we're setting their status in a similar way.

### PR DESCRIPTION
**Pull request by plocket**
In this PR, I have:

* [x] Added tests for any new features or bug fixes (if relevant)
* [x] Added my changes to the CHANGELOG.md at the top, under the "Unreleased" section
* [x] Ensured issues that this PR closes will be [automatically closed](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)

### Reason for this PR

A11y failures were not clear and were using cucumber internals when they shouldn't have. Also restored a11y test for now as a correctly failing test and shortened the aXe file name.

### Links to any solved or related issues

Closes #744. Does not fix da combobox dropdown toggle accessibility issues.

### Any manual testing I have done to ensure my PR is working

None, all done through automated testing.

### CODE REVIEW

File: docassemble/ALKilnTests/data/sources/observation_steps.feature [hunk 1]
```diff
@@ -135,32 +135,33 @@ Scenario: I take a pic
...
-#    | button_continue | True |  |
-#    | buttons_other | button_2 |  |
-#    | buttons_yesnomaybe | True |  |
+# Remove `should be "failed"` when docassemble styles are improved
+# for comboboxes.
+# TODO: Create an actual failing a11y test, maybe using custom html
+@slow @o12 @accessibility @a11y
+Scenario: I check the pages for accessibility
+  Given the final Scenario status should be "failed"
+  And I start the interview at "all_tests"
```
Comment (BryceStevenWilley) [inline docassemble/ALKilnTests/data/sources/observation_steps.feature]:
```suggestion
  Given I start the interview at "all_tests"
```
DA accessibility upstream should be passing now, going to make this quick change and re-run tests to make sure.

**Review by BryceStevenWilley — APPROVED** [review]

**Comment by BryceStevenWilley** [comment]
Looks like there's other things going on now, this error is happening: 

```
1) Scenario: Fail with system error after Step using wrong file name as trigger (attempt 1, retried) # docassemble/ALKilnTests/data/sources/reports.feature:309
   ✔ Before # lib/steps.js:103
   ✔ Given the final Scenario status should be "failed" # lib/steps.js:1084
   ✔ Given the Scenario report should include: # lib/steps.js:1058
       """
       Reference to invalid playground path
       """
   ✖ And I start the interview at "wrong_yaml_filename" # lib/steps.js:163
       AssertionError
           + expected - actual

           -ALKiln could not find any interview question page with the given information

           at Object.load (/home/runner/work/ALKiln/ALKiln/lib/scope.js:797:38)
           at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
           at async World.<anonymous> (/home/runner/work/ALKiln/ALKiln/lib/steps.js:166:3)
   ✖ After # lib/steps.js:1098
       AssertionError: expected '\n---------------\nScenario: Fail with system error after Step using wrong file name as trigger\nTags: @reports @fast @rf20 @error\n---------------\nTrying to load the interview\n\nERROR: On final attempt to load interview, got "ALKiln could not find any interview question page with the given information"\n\n**-- Scenario Failed --**\n' to include 'Reference to invalid playground path'
           at Object.reportIncludesAllExpected (/home/runner/work/ALKiln/ALKiln/lib/scope.js:2996:29)
           at async World.<anonymous> (/home/runner/work/ALKiln/ALKiln/lib/steps.js:1165:29)
```

There's some difference between the test expecting "Reference to invalid playground path", and getting  "ALKiln could not find any interview question page with the given information". I'm not sure how those are different though.

**Comment by plocket** [comment]
Something went wrong with the rebase and axe changes came in as well. Would appreciate your eye, @BryceStevenWilley

**Comment by BryceStevenWilley** [comment]
I'm not sure what's wrong, the rebase looks a little messy, but I redid it in https://github.com/SuffolkLITLab/ALKiln/compare/rebased_744?expand=1, and it looks the exact same. I think that diff is what you added for this PR?

**Comment by plocket** [comment]
Brain freeze, we're good 👍

File: lib/steps.js [hunk 2]
```diff
@@ -1214,6 +1217,19 @@ After(async function(scenario) {
...
   // Save the report as in the Scenario folder
   fs.writeFileSync( scope.paths.scenario_report, report );
+
+  if ( changeable_test_status === `FAILED` ) {
+    // If the result object status wasn't failure and we still want
+    // the tests to fail, as with a11y tests, we can't mess with the
+    // cucumber result.status object because it will cause unwanted
+    // errors. We have to throw an error here instead.
+    if ( scenario.result.status !== `FAILED` ) {
+      throw new Error(`The test failed. See the report for more information.`);
```
Comment (plocket) [inline lib/steps.js:1227]: Add to `.message` here if it doesn't already have a value.

**Pull request merged**